### PR TITLE
ci: file a domain-grouped tracking issue for scheduled PHPUnit failures

### LINKED ISSUE #18707: Nightly PHPUnit failures: auto-filed domain tracking issues and agentic deep triage
Scheduled PHPUnit runs (Nightly → `integration.yml` / `php.yml` / redis, Nightly Major → `integration-major.yml`) fail without follow-up signal: the red run sits in the Actions tab, and extracting which tests failed and which team owns them means reading every shard log by hand.

## Phase 1: domain-grouped tracking issues (#18708)

The scheduled npm audit already solves this shape of problem: `report-scheduled-audit-failures` in `npm-audit-check.yml` aggregates result artifacts and files/updates a marker-tagged tracking issue.

Do the same for PHPUnit:

- upload each failed job's `junit.xml` (already produced by the `phpunit-run` action) as an artifact on scheduled runs
- aggregate them in a report job: failing test → owning domain via the test file's `#[Package]` marker, falling back to the dominant package of the mirrored `src/` directory
- per nightly workflow, create or update one `[nightly]`-prefixed parent tracking issue plus one sub-issue per domain, labeled with the domain label; new failures are appended as comments

This is deliberately an inventory, not a triage: no error-signature clustering, no routing overrides. First consumer is the Nightly Major run tracked in #17978.

<details>
<summary>Phase 1 implementation details</summary>

**Artifact upload.** Every PHPUnit job triggered by the nightlies uploads its `junit.xml` as a `junit-phpunit-*` artifact when it fails on a scheduled run: the `phpunit` matrices in `integration.yml` / `integration-major.yml`, the `unit`/`migration` suites in `php.yml`, and the `redis` job in `nightly.yml` (which now runs with `--log-junit`). Jobs that die before PHPUnit reports produce no artifact (`if-no-files-found: ignore`).

**Report job.** `nightly.yml` and `nightly-major.yml` each get one `report-phpunit-failures` job, gated to `schedule` events and red PHPUnit arms. It calls the reusable workflow `report-phpunit-failures.yml`, which downloads the run's junit artifacts, runs the aggregation script, and applies the issue payload via `actions/github-script`.

**Issue structure.** The parent issue carries the run link and a per-domain overview; the test lists live in the domain sub-issues (linked via the sub-issues REST API, as in `epic-subissues.yml`). Every issue body starts with an HTML marker (`<!-- nightly-phpunit-failures -->`, `<!-- nightly-phpunit-failures:domain/checkout -->`, ...); title plus marker is the dedupe key, so while an issue is open, subsequent nights comment on it instead of filing again. Labels are only set at creation, so manual label changes stick. Tests that resolve to no known package key land in a "needs manual routing" sub-issue without a label.

**Domain resolution.** `#[Package]` marker of the test file, then the dominant package of the mirrored `src/` directory (`tests/{integration,unit}/` maps to `src/`, `tests/migration/<Bundle>/` maps to `src/<Bundle>/Migration/`). The package-key to label mapping mirrors `.agents/skills/sw-triage/references/DOMAINS.md`, which now names the script in its keep-in-sync maintainer note.

**Limits and edge cases.** Test lists are capped at 40 entries per domain. Duplicate failures across shards or matrix variants (e.g. the redis job's image matrix) are deduped by `Class::testName`. A red arm without any junit reports (lint/phpstan failure in `php.yml`, a shard crash during setup) still updates the parent issue with a pointer to the run logs. The LTS nightlies (6.6.x re-dispatch) run as `workflow_dispatch` and never report.

**Deep triage.** `.agents/skills/nightly-triage/SKILL.md` now instructs the triage pass to start from the auto-filed sub-issues and update them (comment, re-route, close duplicates) instead of filing a parallel set.

</details>

## Phase 2: agentic deep triage, sw-nightly (#18715)

The phase 1 issues are an inventory: failing tests grouped by their test files' `#[Package]` markers, no clustering and no routing overrides. The analysis layer (collapse the list into root causes, route each cluster to the owner of the breaking change) is manual: the `nightly-triage` skill.

Run that analysis unattended, in the same fashion as `sw-triage` and `sw-bugfixer`:

- a `qi/sw-nightly` label (or a `/sw-nightly` comment, or a manual dispatch) on a nightly tracking issue triggers a read-only gh aw workflow
- the agent clusters the issue's failing tests by error signature, checks them against the known-cluster and flaky catalogues, locates the breaking change, and emits a structured `NightlyTriageOutput` JSON artifact; it cannot label, comment, or write
- a deterministic processor validates the artifact (shape, closed label catalogue, secret scan) and posts the cluster analysis as a comment on the tracking issue, including re-route suggestions for confirmed root causes whose owner differs from the issue's labels

The rubric is single-sourced in `.github/aw/shared/sw-nightly-policy.md`, shared between the interactive skill and the unattended twin. Local Docker reproduction stays interactive-only; unattended runs mark unverified clusters as "mechanism TBD", which the rubric already treats as keep-with-test-owner.

Rollout: manual labeling first. Once trusted, the phase 1 reporter can attach `qi/sw-nightly` at sub-issue creation to run the analysis automatically for every new domain failure (one line in `report-phpunit-failures.yml`).

<details>
<summary>Phase 2 implementation details</summary>

**Trigger and safety rails.** `.github/workflows/sw-nightly.md` mirrors `sw-triage.md`: Sonnet-tier engine pin, `max-turns: 50`, `timeout-minutes: 20`, `permissions: read-all`, `QUALITY_INITIATIVE_ANTHROPIC_API_KEY` remap, threat-detection on the upload, and a deterministic `nightly-context` artifact carrying issue and run number. The GitHub toolsets add `actions` so the agent can read the failing run's job logs (`get_job_logs`).

**Output contract.** `NightlyTriageOutput`: a `summary` plus 1-10 clusters, each with signature, root cause, confidence (`confirmed | plausible | mechanism-tbd`), `owner_label` from the DOMAINS.md catalogue (or null), `known_cluster` / `flaky_or_environmental` flags, member tests (1-30), provenance-prefixed evidence quotes, and related issue/PR references. Every failing test appears in exactly one cluster.

**Validation.** `.github/bin/js/validate-sw-nightly-output.ts` (plus `node --test` unit tests) enforces the exact shape, the closed label catalogue, all count/length limits, and the same secret-pattern scan as the triage validator (patterns duplicated with a keep-in-sync note; the triage validator executes at top level and cannot be imported).

**Processor.** `.github/workflows/process-sw-nightly-result.yml` triggers on `workflow_run` completion, validates, then creates or updates a marker-tagged comment on the tracking issue: summary, one section per cluster with confidence/owner/flags table, test list and evidence in a collapsed block, and a re-route suggestion when a confirmed cluster's owner label is not on the issue.

**Skill wiring.** The rubric moved to `.github/aw/shared/sw-nightly-policy.md`; `.agents/skills/nightly-triage/SKILL.md` references it and keeps only interactive-mode specifics (Docker reproduction, issue filing with user approval). The skill is now explicit-only (`disable-model-invocation: true`), per the CI-twin convention in `coding-guidelines/core/agent-skills.md`.

</details>

### PR DESCRIPTION
**Pull request by nfortier-shopware**
### 1. Why is this change necessary?

Scheduled PHPUnit runs (Nightly, Nightly Major) fail without follow-up signal; extracting the failing tests and their owning domains from the shard logs is manual work.

### 2. What does this change do, exactly?

- PHPUnit jobs that fail on a scheduled run upload their `junit.xml` as an artifact.
- A report job at the end of `nightly.yml` / `nightly-major.yml` aggregates those artifacts and files (or updates) a parent tracking issue plus one sub-issue per owning domain, labeled with the domain label.
- Domains are resolved from the test files' `#[Package]` markers by the new script `.github/bin/js/report-phpunit-nightly-failures.ts`, covered by `node --test` unit tests.

Implementation details are in #18707.

### 3. Describe each step to reproduce the issue or behaviour.

1. A scheduled Nightly / Nightly Major run has failing PHPUnit jobs.
2. Each failed job uploads its junit report; the report job at the end of the nightly workflow aggregates them.
3. A parent tracking issue and one labeled sub-issue per domain are created (or, when open issues with the same title and body marker exist, today's failures are added to them as comments).

Locally: `cd .github/bin/js && node --test`.

### 4. Please link to the relevant issues (if any).

relates #18707 (phase 1)

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [x] I have written or adjusted the documentation and agent skills according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by BrocksiNet — APPROVED** [review]

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -349,6 +349,15 @@ jobs:
...
           coverage-flags: phpunit-${{ matrix.suite }}
           codecov-token: ${{ secrets.CODECOV_TOKEN }}
+      - name: Upload junit report for the failure tracking issue
+        if: failure() && github.event_name == 'schedule'
+        uses: actions/upload-artifact@043fb46d1a93c77aae656e7c1c64a875d1fc6a0a # v7.0.1
+        with:
+          name: junit-phpunit-${{ matrix.suite }}
+          path: junit.xml
+          # a job that fails before PHPUnit reports (setup, crash) has no junit.xml
+          if-no-files-found: ignore
```
Comment (h1k3r) [inline .github/workflows/php.yml:374]: nit: this protects agains failing worflow, which is good, but missing file could mean error in configuration, so file is not delivered. Do we protect against that?

File: .github/workflows/report-phpunit-failures.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# the PHPUnit jobs of the run (redis, integration.yml, integration-major.yml, php.yml)
+# upload the `junit-phpunit-*` artifacts this consumes.
+
+on:
+  workflow_call:
+    inputs:
+      report-title:
+        type: string
+        required: true
+        description: "Tracking-issue title; together with the body marker it is the dedupe key"
```
Comment (h1k3r) [inline .github/workflows/report-phpunit-failures.yml:14]: What is the upper limit of parent issues created per day with a current setup? One for nightly and one per major?

**Review by h1k3r — APPROVED** [review]

File: .github/workflows/php.yml (hunk 1 above)
Comment (nfortier-shopware) [reply .github/workflows/php.yml:374]: 🤖 says it does (partially) : as it runs whenever a PHPUnit arm failed, so zero delivered reports still files the tracking issue with an explicit warning instead of silently showing nothing

It extended the reporting with
```
'No junit reports were produced although PHPUnit jobs failed. Either the failure happened outside PHPUnit ' +
        '(setup step, crash before reporting), or the junit wiring drifted (`--log-junit` path, `junit-phpunit-*` ' +
        'artifact pattern) and reports are silently no longer delivered. If this note shows up on consecutive runs, ' +
        'treat it as a defect of this reporting pipeline, not as an empty inventory. Check the run logs.'
``

File: .github/workflows/report-phpunit-failures.yml (hunk 2 above)
Comment (nfortier-shopware) [reply .github/workflows/report-phpunit-failures.yml:14]: 2 at most (1 per dedupe key: nightly + nightly-major, each scheduled once daily)

But since red pipeline runs comment on the existing open parent and a new one is only created after the previous, it should be less

**Review by fruppel — APPROVED** [review]
